src/lib/services/feature-toggle-service.ts

```diff
@@ -718,12 +718,22 @@ class FeatureToggleService {
         const strategy = await this.featureStrategiesStore.getStrategyById(
             strategyId,
         );
-        return {
+
+        const segments = await this.segmentService.getByStrategy(strategyId);
+        let result: Saved<IStrategyConfig> = {
             id: strategy.id,
             name: strategy.strategyName,
             constraints: strategy.constraints || [],
             parameters: strategy.parameters,
         };
+
+        if (segments && segments.length > 0) {
+            result = {
+                ...result,
+                segments: segments.map((segment) => segment.id),
+            };
+        }
```

Comment:
Is there a test that validates that this gives the correct result?

Reply:
Also, if there are no segments defined, should we return an empty array?

Reply:
I added one here https://github.com/ivarconr/unleash-enterprise/pull/138

Reply:
@andreas-unleash It seems this is not reflected in the specification, at least I can't see a spec update for this data structure.